bip-0360.mediawiki

```diff
@@ -567,60 +487,91 @@ procedure specified in BIP-341 to maintain compatibility with Taproot signatures
+=== Transaction Size and Fees ===
 
-4. Ensure that the signature algorithm used matches the expected lengths for NIST Level I security, and is supported by
-the implementation.
+Equivalent P2QRH and P2TR outputs are always the same size. P2QRH inputs can be slightly larger or smaller than
+their equivalent P2TR inputs. Let's consider the cases:
 
-=== Compatibility with BIP-141 ===
+'''P2TR key-spend''' P2QRH inputs will be larger than P2TR inputs when the P2TR output would have been spent via the key-spend path.
+P2QRH quantum resistance comes from removing the P2TR key-spend path. Consequently it cannot make use of taproot's optimization
+where P2TR key-spends do not require including a merkle path in the P2TR input. If the Merkle tree only has a single tapleaf, 
+no Merkle path is needed in the control block giving us a 1 byte control block.
 
-By adhering to the SegWit transaction structure and versioning, P2QRH outputs are compatible with existing transaction
-processing rules. Nodes that do not recognize SegWit version 3 will treat these outputs as anyone-can-spend but, per
-[https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki BIP-141], will not relay or mine such transactions.
+P2QRH witness (103 bytes):
+<source>
+[count] (1 byte), # Number of elements in the witness
+[size] signature  (1 + 64 bytes = 65 bytes),
+tapleaf script = [size] [OP_PUSHBYTES_32, 32 byte public key, OP_CHECKSIG] (1 + 1 + 32 + 1 bytes = 35 bytes),
+control block = [size] [control byte]  (1 + 1 = 2 bytes)
+</source>
 
-=== Usage Considerations ===
+P2TR key-spend witness (66 bytes):
+<source>
+[count] (1 byte), # Number of elements in the witness
+[size] signature (1 + 64 bytes = 65 bytes)
+</source>
 
-==== Transaction Size and Fees ====
+Thus, the P2QRH input would be 103 - 66 = 37 bytes larger than a P2TR key-spend input.
 
-Quantum-resistant signatures are significantly larger than traditional signatures, increasing transaction size and the
-fees required. Users and wallet developers should be aware of this and plan accordingly.
+If the Merkle tree has more than a single tapleaf, then the Merkle path must be included in
+the control block.
+P2QRH witness (103+32*m bytes)
+<source>
+[count] (1 byte), # Number of elements in the witness
+[size] signature  (64 + 1 bytes = 65 bytes),
+tapleaf script = [size] [OP_PUSHBYTES_32, 32 byte public key, OP_CHECKSIG] (34 + 1 bytes = 35 bytes),
+control block = [size] [control byte, 32 * m byte Merkle path]  (1 + 1 + 32 * m = 2 + 32 * m bytes)
+</source>
 
-For example, for CRYSTALS-Dilithium Level I, a single public key is 1,312 bytes, and a signature is 2,420 bytes, resulting in a substantial increase over current
-ECDSA or Schnorr signatures.
+For a Merkle path of length m, it would add an additional ~32 * m bytes to the P2QRH input<ref>If the number is large enough we will the compact size will need 2 bytes, increasing the size by 1 byte.</ref>.
```

Comment:
s/we will//

It's probably worth being specific. If m > 7, then the compact size will be 2 bytes. (m = 7 means the control block will be 32 * 7 + 1 bytes = 225 bytes, m = 8 means the control block will be 32 * 8 + 1 bytes = 257 bytes)

Reply:
haha you're totally right, good catch!